garage/envs/mujoco/mujoco_env.py

```diff
@@ -116,12 +116,34 @@ def reset_mujoco(self, init_state=None):
             self.sim.data.ctrl[:] = self.init_ctrl
         else:
             start = 0
-            for datum_name in ["qpos", "qvel", "qacc", "ctrl"]:
-                datum = getattr(self.sim.data, datum_name)
-                datum_dim = datum.shape[0]
-                datum = init_state[start:start + datum_dim]
-                setattr(self.sim.data, datum_name, datum)
-                start += datum_dim
+
+            # update qpos
+            datum = self.sim.data.qpos
+            datum_dim = datum.shape[0]
+            datum = init_state[start:start + datum_dim]
+            self.sim.data.qpos[:] = datum
+            start += datum_dim
```

Comment:
I think this code would be most impervious to refactoring and shorter if you just pre-calculated the offsets (`start`). Right now, if these lines get rearranged, the final array will end up with the wrong ordering (very difficult to debug).

Something like:
```python
datums = [
    self.sim.data.qpos,
    self.sim.data.qvel,
    self.sim.data.qacc,
    self.sim.data.ctrl,
]
dims = [d.shape[0] for d in datums]
offsets = [0] + list(np.cumsum(dims)[:-1])

for datum, dim, offset in zip(datums, dims, offsets):
    datum[:] = init_state[offset:offset + dim] 
```

that's it! no metaprogramming!

Reply:
Thank you so much for your pointers, @ryanjulian and I tested your solution out! it's working 👍